ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -271,15 +271,17 @@ public void setAutoCapitalize(ReactEditText view, int autoCapitalize) {
 
   @ReactProp(name = "keyboardType")
   public void setKeyboardType(ReactEditText view, @Nullable String keyboardType) {
+    int flagsToUnset = 0;
     int flagsToSet = 0;
     if (KEYBOARD_TYPE_NUMERIC.equalsIgnoreCase(keyboardType)) {
-      flagsToSet = InputType.TYPE_CLASS_NUMBER;
+      flagsToUnset = InputType.TYPE_MASK_CLASS | InputType.TYPE_MASK_FLAGS;
```

Comment:
I would agree. However, all the other flags that are set are prefix with TYPE_TEXT_, which implies they for when the input type class is text. But here we are changing the input type class to number, so it would seem that those text flags are no long applicable or even appropriate.